tests/unit/aspect.ts

```diff
@@ -171,6 +171,41 @@ registerSuite({
 			}
 		},
 
+		'.on()': {
+			'advising function returns undefined, returns original result'() {
+				const aspectStub1 = sinon.stub();
+
+				aspect.on(obj, 'method', aspectStub1);
+
+				assert.strictEqual(obj.method(0), 1);
+				assert.isTrue(aspectStub1.calledAfter(methodSpy));
+			},
+
+			'advising function returns defined values, returns advising function result'() {
+				const aspectStub1 = sinon.stub().returns(2);
+
+				aspect.on(obj, 'method', aspectStub1);
+
+				assert.strictEqual(obj.method(0), 2);
```

Comment:
As I alluded to in my previous comment, `aspect.on` is equivalent to Dojo 1's `aspect.after(obj, method, func, true)`, while `aspect.after` here is the same as Dojo 1's `aspect.after(obj, method, func, false)`.

IIRC @bryanforbes came up with the idea of separating its API surface, since it really is more akin to an event handler (in that it passes the original arguments along to the advising function, not the return value) than to AOP.

I believe cujojs/meld has the [same idea](https://github.com/cujojs/meld/blob/master/docs/api.md#meldon).

Your concern over confusion over which `on` to use is valid.  At a technical level it comes down to differentiating event emitters and events from generic objects using methods as join points.